php/Controllers/SettingsController.php

```diff
@@ -745,96 +673,117 @@ public function plugin_setting_section_cb() {
+        echo $this->twig->render('Partials/Paragraph.html.twig', [
+            'text' => __('In order to use the plugin, configure the API settings accordingly. Once your profile has been retrieved and an organisation has been selected, you can optionally process any existing posts, pages and images to populate the Coyote instance.', WordPressPlugin::I18N_NS),
+        ]);
     }
 
     public function noop_setting_section_cb() {}
 
     public function api_endpoint_cb() {
-        ?>
-        <input name="coyote_api_endpoint" id="coyote_api_endpoint" type="text" value="<?= esc_url(pluginConfiguration::getApiEndPoint()) ?>" size="50" aria-describedby="coyote_api_endpoint_hint"/>
-        <p id="coyote_api_endpoint_hint"><?php _e('The endpoint for your Coyote instance, e.g. "https://staging.coyote.pics".', WordPressPlugin::I18N_NS); ?></p>
-        <?php
+        echo $this->twig->render('Partials/InputText.html.twig', [
+            'name'                  => 'coyote_api_endpoint',
+            'label'                 => __('The endpoint for your Coyote instance, e.g. "https://staging.coyote.pics".', WordPressPlugin::I18N_NS),
+            'size'                  => 50,
+            'value'                 => esc_url(pluginConfiguration::getApiEndPoint())
+        ]);
     }
 
     public function api_token_cb() {
-        ?>
-        <input name="coyote_api_token" id="coyote_api_token" type="text" value="<?= sanitize_text_field(pluginConfiguration::getApiToken()); ?>" size="30" aria-describedby="coyote_api_token_hint"/>
-        <p id="coyote_api_token_hint"><?php _e('The API token associated with your Coyote account.', WordPressPlugin::I18N_NS); ?></p>
-        <?php
+        echo $this->twig->render('Partials/InputText.html.twig', [
+            'name'                  => 'coyote_api_token',
+            'label'                 => __('The API token associated with your Coyote account.', WordPressPlugin::I18N_NS),
+            'size'                  => 30,
+            'value'                 => sanitize_text_field(pluginConfiguration::getApiToken())
+        ]);
     }
 
     public function api_metum_cb() {
-        ?>
-        <input name="coyote_api_metum" id="coyote_api_metum" type="text" value="<?= sanitize_text_field(pluginConfiguration::getMetum()); ?>" size="20" aria-describedby="coyote_api_metum_hint"/>
-        <p id="coyote_api_metum_hint"><?php _e('The metum used by the API to categorise image descriptions, e.g. "Alt".', WordPressPlugin::I18N_NS); ?></p>
-        <?php
+        echo $this->twig->render('Partials/InputText.html.twig', [
+            'name'                  => 'coyote_api_metum',
+            'label'                 => __('The metum used by the API to categorise image descriptions, e.g. "Alt".', WordPressPlugin::I18N_NS),
+            'size'                  => 20,
+            'value'                 => sanitize_text_field(pluginConfiguration::getMetum())
```

Comment:
Is the sanitize call still necessary when it's going into a template engine?

Reply:
Twig does not seem to have a filter to sanitize strings, only to esc certain things like html, js, css.

Reply:
Ofcourse it does have filters to sanitize a string for certain things like trim and striptags, but it does not seem to be as complete as what this method does.